clippy_lints/src/strings.rs

```diff
@@ -208,16 +214,38 @@ fn is_string(cx: &LateContext<'_>, e: &Expr<'_>) -> bool {
     is_type_lang_item(cx, cx.typeck_results().expr_ty(e).peel_refs(), LangItem::String)
 }
 
-fn is_add(cx: &LateContext<'_>, src: &Expr<'_>, target: &Expr<'_>) -> bool {
+fn sugg_push_or_pushstr(cx: &LateContext<'_>, s: &Expr<'_>) -> String {
+    if let ExprKind::Lit(Spanned {
+        node: LitKind::Str(symbol, ..),
+        ..
+    }) = s.kind
+    {
+        match symbol.to_string().chars().count() {
```

Comment:
You can add a `.take(2)` before `.count()` to not count all the codepoints.

Also, if the string literal is the empty string, you could maybe suggest an alternative to the current `s.push_str("")`.